mat64/inner.go

```diff
@@ -29,26 +29,65 @@ func Inner(x []float64, A Matrix, y []float64) float64 {
 	switch b := A.(type) {
 	case RawSymmetricer:
 		bmat := b.RawSymmetric()
```

Comment:
Can't do `case RawSymmetricer, RawTriangular:`. We can do separate cases and send the results to a func.